src/legacy/ui/public/saved_objects/components/saved_object_finder.tsx

```diff
@@ -248,73 +328,169 @@ class SavedObjectFinder extends React.Component<SavedObjectFinderProps, SavedObj
             data-test-subj="savedObjectFinderSearchInput"
           />
         </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiFilterGroup>
+            <EuiToolTip position="top" content={this.getCurrentSortTooltip()}>
+              <EuiButtonEmpty
+                size="s"
+                color="text"
+                data-test-subj="savedObjectFinderSortButton"
+                onClick={() =>
+                  this.setState(({ sortDirection, filter }) => ({
+                    sortDirection: this.getNextDirection(sortDirection, Boolean(filter)),
+                  }))
+                }
+              >
+                <FormattedMessage
+                  id="common.ui.savedObjects.finder.sortByLabel"
+                  defaultMessage="Sort: "
+                />
+                <EuiIcon type={this.getCurrentSortIcon()} />
+              </EuiButtonEmpty>
+            </EuiToolTip>
+            {this.props.showFilter && availableSavedObjectMetaData.length > 1 && (
+              <EuiPopover
+                id="popover"
+                panelClassName="euiFilterGroup__popoverPanel"
+                panelPaddingSize="none"
+                isOpen={this.state.filterOpen}
+                closePopover={() => this.setState({ filterOpen: false })}
+                button={
+                  <EuiButtonEmpty
```

Comment:
This one as well. You can just use a `<EuiFilterButton>`.

```jsx
<EuiFilterButton
        iconType="arrowDown"
        onClick={}
        isSelected={this.state.isPopoverOpen}
        numFilters={items.length} // This number should equal the number total
        hasActiveFilters={true} // This should equal true if they've selected any types
        numActiveFilters={2} // This number should equal the number selected
      >
          Types
</EuiFilterButton>
```

Reply:
EuiFilterButton looks much better here although there seems to be a bug in the current EUI version with numActiveFilters not working correctly:
```
Warning: React does not recognize the `numActiveFilters` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `numactivefilters` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
```

I worked around it by just using numFilters here, we can adjust this when EUI behaves correctly.

Reply:
Ahh it's because your branch is still on EUI v. `7.1.0` and that was added in `9.0`. Kibana master is now at EUI `9.0.2`. You probably want to update from master.